include/seastar/core/semaphore.hh

```diff
@@ -518,6 +561,14 @@ public:
     explicit operator bool() const noexcept {
         return _n != 0;
     }
+
+    bool broken() const noexcept {
```

Comment:
We can have such a method (I think that `check()` would be enough, similar to `abort_source::check`.
But I wouldn't add it to methods that don't need to access the semaphore as it might break @gleb-cloudius's use cases ;-)

Reply:
See saga in https://github.com/scylladb/scylladb/pull/12725